src/appleseed.shaders/include/appleseed/material/as_hazy_gloss.h

```diff
@@ -0,0 +1,335 @@
+}
+
+
+
+closure color as_hazy_gloss
+(
+    vector Nn,
+    vector tangent,
+    float in_specular_roughness,
+    float in_specular_haziness,
+    float in_specular_haze_extent,
+    float in_anisotropy_amount,
+    float in_ior
+)
+{
+    float reflectivity = sqr((in_ior - 1.0) / (in_ior + 1.0));
+    float alpha_n = sqr(in_specular_roughness); // map to microfacet alpha
+    float haze_extent = 10.0 * in_specular_haze_extent; // map to a larger range
+
+    float eta;
+    float beta;
+    float alpha_w;
+    haze_mapping_to_ggx(reflectivity, alpha_n, in_specular_haziness, haze_extent, eta, beta, alpha_n, alpha_w);
+
+    float roughness_2 = sqrt(alpha_w);
+    float new_ior = (1+sqrt(eta))/(1-sqrt(eta));
+
+    return as_double_ggx_gloss(
+        Nn,
+        tangent,
+        in_specular_roughness,
+        roughness_2,
+        beta,
+        in_anisotropy_amount,
+        new_ior
+    );
+}
+
+
+
+closure color as_hazy_glass
+(
+    vector Nn,
+    vector tangent,
+    color coating_transmittance,
+    color in_specular_color,
+    color in_scaled_refraction_tint,
+    float in_specular_roughness,
+    float in_specular_haziness,
+    float in_specular_haze_extent,
+    float in_anisotropy_amount,
+    float in_ior,
+    color in_absorption_color,
+    float in_absorption_depth
+)
+{
+    float reflectivity = sqr((in_ior - 1.0) / (in_ior + 1.0));
+    float alpha_n = sqr(in_specular_roughness); // map to microfacet alpha
+    float haze_extent = 10.0 * in_specular_haze_extent; // map to a larger range
+
+    float eta;
+    float beta;
+    float alpha_w;
+    haze_mapping_to_ggx(reflectivity, alpha_n, in_specular_haziness, haze_extent, eta, beta, alpha_n, alpha_w);
+
+    float roughness_2 = sqrt(alpha_w);
+    float new_ior = (1+sqrt(eta))/(1-sqrt(eta));
+
+    return as_double_ggx_glass(
+        Nn,
+        tangent,
+        coating_transmittance,
+        in_specular_color,
+        in_scaled_refraction_tint,
+        in_specular_roughness,
+        roughness_2,
+        beta,
+        in_anisotropy_amount,
+        new_ior,
+        in_absorption_color,
+        in_absorption_depth
+    );
+}
+
+
+closure color as_hazy_metal
+(
+    vector Nn,
+    vector tangent,
+    color in_specular_color,
+    color in_face_tint,
+    color in_edge_tint,
+    float in_specular_roughness,
+    float in_specular_haziness,
+    float in_specular_haze_extent,
+    float in_anisotropy_amount
+)
+{
+    float reflectivity = max(in_specular_color);
+    float alpha_n = sqr(in_specular_roughness); // map to microfacet alpha
+    float haze_extent = 10.0 * in_specular_haze_extent; // map to a larger range
+
+    float eta;
+    float beta;
+    float alpha_w;
+    haze_mapping_to_ggx(reflectivity, alpha_n, in_specular_haziness, haze_extent, eta, beta, alpha_n, alpha_w);
+
+    float roughness_2 = sqrt(alpha_w);
+
+    return eta/reflectivity * as_double_ggx_metal(
+        Nn,
+        tangent,
+        in_face_tint,
```

Comment:
The new reflectance value you compute in the conductor haze mapping goes here.
(r_max * chromaVec) in the paper BRDF explorer sample code.